implementation.md

```diff
@@ -1554,3 +1554,272 @@ In the `Video` and `Audio` objects, sellers must only include one form of floor
+
+#### 7.12.4.3 Derrived IDs Across Multiple Properties or Devices
+##### Example 5: SSP matches their own ID to an email-based hash to provide a universal ID to the DSP in the buyeruid field (based on prior arrangement to put it in buyeruid)
+```
+{
+"user": {
+    "buyeruid": "fac13741-0648-436a-88cf-aceafdf45c9a",
+    "eids": [
+        {
+            "source": "sspdomain.com”,
+            "mm": 1,
+            "uids": [
+                {
+                    "id": "3d89748d-74bb-44f1-9908-298090dc2944",
+		    "atype": "1",
+                }
+            ]
+        },
+        {
+            "matcher": "idvendor.com",
+	    "inserter": "sspdomain.com",
+            "source": "universalid.com”,
+            "mm": 4,
+            "uids": [
+                {
+                    "id": "fac13741-0648-436a-88cf-aceafdf45c9a",
+		    "atype": "3",
+                }
+            ]
+        }
+    ]
+}
+}
+```
+##### Example 6: On a CTV request, the publisher also provides the IFA and an alternative ID of an Apple device within that same household, based on authenticated logins across apps
+
+```
+{
+"user": {
+    "eids": [
+        {
+	    "matcher": "app-pub.com",
+	    "inserter": "app-pub.com",
+	    "source": "apple.com”,
+            "mm": 3,
+            "uids": [
+                {
+                    "id": "fac13741-0648-436a-88cf-aceafdf45c9a",
+		    "atype": "2",
+                }
+            ]
+        },
+        {
+            "matcher": "app-pub.com",
+	    "inserter": "app-pub.com",
+            "source": "alt-id.org”,
+            "mm": 3,
+            "uids": [
+                {
+                    "id": "e7f8c5f2-6b4e-4565-8b0a-53d8b730143e",
+		    "atype": "3",
+                }
+            ]
+        }
+    ]
+}
+}
+“device”: {
+    "ifa": "df3a42f4-a388-43e9-a691-b0d308c963b4",
+}
+```
+
+##### Example 7: On a CTV request, the publisher provides the IFA of the TV, but also adds the IFA of an Apple device within that same household to the extended IDs array, based on the household’s IP and a device graph
+```
+{
+"user": {
+    "eids": [
+        {
+	    "matcher": "graphvendor.com",
+	    "inserter": "app-pub.com",
+	    "source": "apple.com”,
+            "mm": 4,
+            "uids": [
+                {
+                    "id": "fac13741-0648-436a-88cf-aceafdf45c9a",
+		    "atype": "3",
+                }
+            ]
+        }
+}
+“device”: {
+    "ifa": "df3a42f4-a388-43e9-a691-b0d308c963b4",
+}
+```
+#### 7.12.4.4 Session IDs
+##### Example 8: FAST (free ad-supported TV) app developer provides an ID to the SSP for a non-authenticated user, on a request with no OS-provided device ID
+```
+{
+"user": {
+    "eids": [
+        {
+	    "inserter": "app-pub.com",
+	    "source": "app-pub.com”,
+            "mm": "6",
+            "uids": [
+                {
+                    "id": "fac13741-0648-436a-88cf-aceafdf45c9a",
+		    "atype": "2",
```

Comment:
Pretty sure this shouldn't be `atype` 1 or 2 or 3. It doesn't match any of the definitions. Do we need to add a new atype for a session ID? `"atype": 4`?